tests/suites/client/HandshakeTest.php

```diff
@@ -72,7 +72,7 @@ public function testHandshakeResponse(): void
         $response = $client->getHandshakeResponse();
         $this->assertInstanceOf(Response::class, $response);
         $this->assertEquals(101, $response->getStatusCode());
-        $this->assertEquals('Switching Protocols', $response->getReasonPhrase());
+        $this>assertContains($response->getReasonPhrase(), ['Switching Protocols', '', ' ']);
```

Comment:
Keep the exact match here - it should be exact.

Instead add a test in `tests/suites/http/HttpHandlerTest.php`. There's a method `testPullResponse` you can copy and adapt to test missing reason phrase. Note that the code resolves reason from status code if missing so `getReasonPhrase()` should only return empty string if the status code is non-standard.

Reply:
Thx for your feedback.
I'm not that familiar with phpunit and I'm trying to run the tests locally before pushing any change with `vendor/bin/phpunit vendor/phrity/websocket/tests/suites/client/HandshakeTest.php` but I got this error

```
Fatal error: Trait "Phrity\Net\Mock\Stack\ExpectSocketClientTrait" not found in /vendor/phrity/websocket/tests/suites/client/HandshakeTest.php on line 39
```

Executing `vendor/bin/phpunit -c vendor/phrity/websocket/phpunit.xml` returns

```
Warning: require(/vendor/phrity/websocket/vendor/autoload.php): Failed to open stream: No such file or directory in /vendor/phrity/websocket/tests/bootstrap.php on line 10

```

What is the best way to run the tests locally?

Reply:
You need to run phpunit in the fork you made of this library. 
Place your console in the root of the `websocket-php` (the fork).

If you have `make` installed, you can just run `make test` and it will set things up.
If not you first need to run composer to load the dependencies (a vendor folder will be installed in the fork) and then run `php vendor/bin/phpunit`. The config and bootstrap will be used without extra arguments.

Reply:
Thanks, I didn't noticed that Makefile. 

I'm trying to run the tests inside a Docker container (based on php:8.3-alpine) with `make test`.

```
./vendor/bin/phpunit
PHPUnit 11.5.7 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.16
Configuration: /nostr-php/websocket-php/phpunit.xml

.F.FFFFFFFFF.E.F.EFF.......FFFFFFFF.FFFF.FFFF.FF....FFFFFFFFFFF  63 / 199 ( 31%)
FF....................F........................................ 126 / 199 ( 63%)
.............................FF.......F.......F.FEEEEEEEEEEEEEE 189 / 199 ( 94%)
EEE.EEEFEE                                                      199 / 199 (100%)
```
which give me 24 errors.

It seems no WebSocket server can be initiated.

Reply:
It uses standard php installation, no dependency on non-standard extensions.
If it's only the server failing to init I would guess it can't access sockets.

In this case, you should be able to run the specific test - which do not rely on server being set up.
```
php vendor/bin/phpunit tests/suites/http/HttpHandlerTest.php
```

If that test passes, and you haven't changed anything else, the GitHub build should pass as well.

Reply:
The specific test needs a running websocket 

```
php vendor/bin/phpunit tests/suites/client/HandshakeTest.php
```

But after I reverted my last commit https://github.com/sirn-se/websocket-php/pull/100/commits/5a7443bf3800b7cafc4c4ed956b7f8d76fcdae49, I see the tests are now passed.